revocation/internal/crl/crl.go

```diff
@@ -73,37 +71,67 @@ func CertCheckStatus(ctx context.Context, cert, issuer *x509.Certificate, opts C
 			Result: result.ResultNonRevokable,
 			ServerResults: []*result.ServerResult{{
 				RevocationMethod: result.RevocationMethodCRL,
+				Error:            errors.New("CRL is not supported"),
 				Result:           result.ResultNonRevokable,
 			}},
 			RevocationMethod: result.RevocationMethodCRL,
 		}
 	}
 
+	var (
+		serverResults = make([]*result.ServerResult, 0, len(cert.CRLDistributionPoints))
+		lastErr       error
+		crlURL        string
+	)
+
+	cachedFetcher, err := fetcher.NewCachedFetcher(opts.HTTPClient, opts.CacheClient)
```

Comment:
It is not necessary to create a fetcher each time checking the status.

Can you replace `HTTPClient *http.Client` in `CertCheckStatusOptions` with `fetcher crl.Fetcher`? BTW, don't worry about the cyclic import (if any in the future) since we can always duplicate the interface in different packages.

Reply:
Updated.